templates/types/streaming/nextjs/app/api/embed/route.ts

```diff
@@ -0,0 +1,28 @@
+import { NextRequest, NextResponse } from "next/server";
+import { initSettings } from "../engine/settings";
+import { getPdfDetail } from "./embeddings";
+
+initSettings();
+
+export async function POST(request: NextRequest) {
+  try {
+    const { pdf }: { pdf: string } = await request.json();
+    if (!pdf) {
+      return NextResponse.json(
+        { error: "pdf is required in the request body" },
+        { status: 400 },
+      );
+    }
+    const pdfDetail = await getPdfDetail(pdf);
+    return NextResponse.json(pdfDetail);
```

Comment:
you can directly return `TextNode[]` here, but remove metadata first as the metadata is very big